### DISCUSSION TITLE
Next.js Tutorial - Script component

### DISCUSSION THREAD

**Discussion by mmaksi** [vercel/next.js#45477] [Help] [answered]
While following the official Next.js tutorial, I came across the concept of optimizing scripts here [on this page](https://nextjs.org/docs/api-reference/next/script#strategy). I am supposed to see a log in the developer tool's console but I can see nothing. Is it because of an issue in the Script component or because of the source link itself?

Here is the link for the repo: https://github.com/mmaksi/nextjs-tutorial

**Comment by ciruz** [comment, marked as answer]
Hey,

No script tags inside Head Component => https://nextjs.org/docs/messages/no-script-tags-in-head-component

If you use the ``<Script>`` outside of ``<Head>`` it works fine:

```javascript
//first-post.js

import Head from "next/head";
import Link from "next/link";
import Script from "next/script";

export default function FirstPost() {
  return (
    <>
      <Script
        src="https://connect.facebook.net/en_US/sdk.js"
        strategy="lazyOnload"
        onLoad={() =>
          console.log(`script loaded correctly, window.FB has been populated`)
        }
      />
      <Head>
        <title>First Post</title>
      </Head>

      <h1>First Post</h1>
      <h2>
        <Link href="/">Back to home</Link>
      </h2>
    </>
  );
}
```

![image](https://user-images.githubusercontent.com/987947/216033700-3061fa18-2ab1-4093-91a3-64de1dfc7ee7.png)

**Reply by mmaksi** [reply]
Thank you so much!

**Reply by ciruz** [reply]
Also if you really want to use the FB SDK, you need to initalize it (``FB.init({.....});``), so check this out too:

https://nextjs.org/docs/basic-features/script#inline-scripts
